website/docs/tutorials/08-cairo-1-support/02-project-initialization.md

```diff
@@ -14,100 +16,285 @@ protostar init-cairo1 my-project
+
+```cairo
 mod functions;
 ```
-5. Declare the module in the root of the module tree - `lib.cairo`. After adding, the file contents should look like this:
-```
-mod main;
+
+4. Update the `lib.cairo` file to include `mod1`. It's contents should now look like this
+
+```cairo
+// previous code stays
+// ...
 mod mod1;
 ```
-6. You can import the symbols from `functions.cairo` in tests. For example, in `test_main.cairo`:
-```
-use src::mod1::functions::three;
 
-#[test]
-fn test_numbers() {
-    assert(3 == three(), 'three() == 3');
-}
+If you followed the steps correctly, your new project structure should look like this
+
+```
+my_project/
+├── hello_starknet/
+│   ├── src/
+│   │   ├── business_logic/
+│   │   │   └── utils.cairo
+│   │   ├── contracts/
+│   │   │   └── hello_starknet.cairo
+│   │   ├── mod1/  <------------------- new directory
+│   │   │   └── functions.cairo  <----- new file
+│   │   ├── business_logic.cairo
+│   │   ├── contracts.cairo
+│   │   ├── lib.cairo  <--------------- contents updated
+│   │   └── mod1.cairo  <-------------- new file
+│   └── cairo_project.toml
+├── tests/
+│   ├── test_hello_starknet.cairo
+│   └── test_utils.cairo
+└── protostar.toml
 ```
 
+#### Using added module
+
+You now use your function in the HelloStarknet contract use `my_project::mod1::functions::returns_three()`.
+
 ## The protostar.toml
-Apart from the usual things you can find in `protostar.toml`, there is a `linked-libarires` entry which is used to find cairo 1 modules in tests and building.
-This makes it possible for you to include other modules from your dependencies if they are correct cairo 1 modules (with their own module definition and `cairo_project.toml`).
+
+Apart from the usual things you can find in `protostar.toml`, there is a `linked-libarires` entry which is used to find
+cairo1 modules in tests and building.
+This makes it possible for you to include other modules from your dependencies if they are correct cairo1 modules (with
+their own module definition and `cairo_project.toml`).
 
 ```
 [project]
 protostar-version = "0.9.2"
 lib-path = "lib"
-linked-libraries = ["src"]
+linked-libraries = ["hello_starknet"]
 
 [contracts]
+hello_starknet = ["hello_starknet"]
 ```
 
-:::warning
-`[contracts]` section is not usable right now, since protostar can't build cairo 1 contracts yet 
-:::
\ No newline at end of file
+:::info
+Even though `hello_starknet.cairo` file is defined in the nested directory, we use `"src"` as path to the contract. This
+is necessary for the imports from other modules (like `business_logic`) to work.
+:::
+
+## Using multiple contracts in project
+
+Due to limitations of cairo1 compiler, having multiple contracts defined in the project will cause `protostar build` and
+other commands to fail.
+
+**That is, having projects structured like this is not valid and will not work correctly.**
+
+### ❌ Incorrect multi-contract project structure
```

Comment:
We shouldn't include incorrect schema, I don't see a point in that

Reply:
I think for some users having an example of what cannot be done may be useful. Especially since I think many will initially attempt to create a structure like this.

Reply:
i disagree, no point in showing wrong structure